app/code/community/Ebizmarts/MailChimp/Model/Api/Subscribers.php

```diff
@@ -330,53 +327,64 @@ public function updateSubscriber($subscriber, $updateStatus = false)
         $isAdmin = Mage::app()->getStore()->isAdmin();
         $helper = $this->mcHelper;
         $storeId = $subscriber->getStoreId();
-        $listId = $helper->getGeneralList($storeId);
-        $newStatus = $this->translateMagentoStatusToMailchimpStatus($subscriber->getStatus(), $storeId);
-        $forceStatus = ($updateStatus) ? $newStatus : null;
-        $api = $helper->getApi($storeId);
-        $mergeVars = $this->getMergeVars($subscriber);
-        $md5HashEmail = md5(strtolower($subscriber->getSubscriberEmail()));
-        try {
-            $api->lists->members->addOrUpdate(
-                $listId, $md5HashEmail, $subscriber->getSubscriberEmail(), $newStatus, null, $forceStatus, $mergeVars,
-                null, null, null, null
-            );
-            $subscriber->setData("mailchimp_sync_delta", Varien_Date::now());
-            $subscriber->setData("mailchimp_sync_error", "");
-            $subscriber->setData("mailchimp_sync_modified", 0);
-        } catch (MailChimp_Error $e) {
-            if ($newStatus === 'subscribed' && $subscriber->getIsStatusChanged()) {
-                if (strstr($e->getMailchimpDetails(), 'is in a compliance state')) {
-                    try {
-                        $api->lists->members->update($listId, $md5HashEmail, null, 'pending', $mergeVars);
-                        $subscriber->setSubscriberStatus(Mage_Newsletter_Model_Subscriber::STATUS_UNCONFIRMED);
-                        $message = $helper->__('To begin receiving the newsletter, you must first confirm your subscription');
-                        Mage::getSingleton('core/session')->addWarning($message);
-                    } catch (MailChimp_Error $e) {
+        $subscriptionEnabled = $helper->isSubscriptionEnabled($storeId);
+        if ($subscriptionEnabled) {
+            $listId = $helper->getGeneralList($storeId);
+            $newStatus = $this->translateMagentoStatusToMailchimpStatus($subscriber->getStatus(), $storeId);
+            $forceStatus = ($updateStatus) ? $newStatus : null;
+            try {
```

Comment:
Move the `try/catch` for `getApi` to a separate block (like in app/code/community/Ebizmarts/MailChimp/Model/Api/Orders.php line 249), to avoid nested `try/catch` blocks.